toolkit/tools/pkg/imagecustomizerlib/telemetry.go

```diff
@@ -0,0 +1,78 @@
+package imagecustomizerlib
+
+import (
+	"context"
+	"net"
+	"runtime"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+	"go.opentelemetry.io/otel/trace/noop"
+)
+
+var (
+	Tracer trace.Tracer
+
+	TracerName = "imagecustomizer"
+
+	TelemetryServerAddress = "localhost:4317"
+
+	shutdownFn func(ctx context.Context) error
+)
+
+func InitTelemetry(disableTelemetry bool) error {
+	if disableTelemetry || !checkServerListening() {
+		// Use noop tracer when telemetry is disabled
+		logger.Log.Trace("Disabled telemetry collection")
+		Tracer = noop.NewTracerProvider().Tracer(TracerName)
+		shutdownFn = nil
+		return nil
+	}
+
+	exporter, err := otlptracegrpc.New(context.Background(),
```

Comment:
There is the [autoexport](https://pkg.go.dev/go.opentelemetry.io/contrib/exporters/autoexport) package that allows the OTEL connection config to be controlled via environment variables. That is probably a better solution than hardcoding it.

If `OTEL_TRACES_EXPORTER` is set, then we can initialize the tracer.